packages/plugins/experimental/plugin-inbox/src/components/Mailbox/Mailbox.tsx

```diff
@@ -17,67 +24,182 @@ const byDate =
   ({ created: a = '' }: MessageType, { created: b = '' }: MessageType) =>
     a < b ? -direction : a > b ? direction : 0;
 
-const setMessageProperty = (message: MessageType, property: string, value: any) => {
-  if (!message.properties) {
-    message.properties = {};
-  }
+const updateMessageProperty = (messages: MessageType[], property: string, value: any) => {
```

Comment:
anything mutating messages is deprecated and I'd remove it. There's proposals for editing messages on queues but currently none of it is implemented.

Understand if you want to try to afford some space in the UI for where actions might go but I wouldn't bother trying to retain existing implementations